libnetwork/resolvconf/resolvconf_unix_test.go

```diff
@@ -416,11 +435,788 @@ func TestFilterResolvDNS(t *testing.T) {
+			reqdOptions:   []string{"ndots:0"},
+			expExtServers: []ExtDNSEntry{mke("127.0.0.53", true)},
+			expContent: `nameserver 127.0.0.11
+options ndots:1
+
+# Based on host file: '/etc/resolv.conf' (internal resolver)
+# ExtServers: [host(127.0.0.53)]
+# Overrides: []
+# Option ndots from: host
+`,
+		},
+		{
+			name:          "ndots missing but required",
+			input:         "nameserver 127.0.0.53",
+			reqdOptions:   []string{"ndots:0"},
+			expExtServers: []ExtDNSEntry{mke("127.0.0.53", true)},
+			expContent: `nameserver 127.0.0.11
+options ndots:0
+
+# Based on host file: '/etc/resolv.conf' (internal resolver)
+# ExtServers: [host(127.0.0.53)]
+# Overrides: []
+# Option ndots from: internal
+`,
+		},
+		{
+			name:            "ndots host, override and required",
+			input:           "nameserver 127.0.0.53",
+			reqdOptions:     []string{"ndots:0"},
+			overrideOptions: []string{"ndots:2"},
+			expExtServers:   []ExtDNSEntry{mke("127.0.0.53", true)},
+			expContent: `nameserver 127.0.0.11
+options ndots:2
+
+# Based on host file: '/etc/resolv.conf' (internal resolver)
+# ExtServers: [host(127.0.0.53)]
+# Overrides: [options]
+# Option ndots from: override
+`,
+		},
+		{
+			name:          "Extra required options",
+			input:         "nameserver 127.0.0.53\noptions trust-ad",
+			reqdOptions:   []string{"ndots:0", "attempts:3", "edns0", "trust-ad"},
+			expExtServers: []ExtDNSEntry{mke("127.0.0.53", true)},
+			expContent: `nameserver 127.0.0.11
+options trust-ad ndots:0 attempts:3 edns0
+
+# Based on host file: '/etc/resolv.conf' (internal resolver)
+# ExtServers: [host(127.0.0.53)]
+# Overrides: []
+# Option ndots from: internal
+`,
+		},
+		{
+			name:          "Bad internal nameserver",
+			input:         "nameserver 127.0.0.53",
+			intNameServer: "1.2.3.4.5",
+			expErr:        "ParseAddr(\"1.2.3.4.5\"): IPv4 address too long",
+		},
+	}
+
+	for _, tc := range testcases {
+		t.Run(tc.name, func(t *testing.T) {
+			tc := tc
+			rc, err := readResolvConf("/etc/resolv.conf", bytes.NewBuffer([]byte(tc.input)))
+			assert.NilError(t, err)
+
+			if tc.intNameServer == "" {
+				tc.intNameServer = "127.0.0.11"
+			}
+			if len(tc.overrideNS) > 0 {
+				rc.OverrideNameServers(tc.overrideNS)
+			}
+			if len(tc.overrideOptions) > 0 {
+				rc.OverrideOptions(tc.overrideOptions)
+			}
+			err = rc.TransformForIntNS(tc.ipv6, tc.intNameServer, tc.reqdOptions)
+			if tc.expErr != "" {
+				assert.Check(t, is.ErrorContains(err, tc.expErr))
+				return
+			}
+			assert.NilError(t, err)
+
+			d := t.TempDir()
+			path := filepath.Join(d, "resolv.conf")
+			err = rc.WriteFile(path, "", 0644)
+			assert.NilError(t, err)
+
+			content, err := os.ReadFile(path)
+			assert.NilError(t, err)
+			// fmt.Print(string(content))
+			assert.Check(t, is.DeepEqual(string(content), tc.expContent))
+			assert.Check(t, is.DeepEqual(rc.ExtNameServers(), tc.expExtServers,
+				cmp.Comparer(func(a, b ExtDNSEntry) bool { return a == b })))
```

Comment:
You shouldn't have to supply a custom Comparer unless you need to customize the definition of equality to be something other than `==` (a.k.a. `(reflect.Value).Equal`)

Reply:
Without it ...

```
    resolvconf_unix_test.go:1143: assertion failed: cannot handle unexported field at {[]resolvconf.ExtDNSEntry}[0].Addr.addr:
        	"net/netip".Addr
        consider using a custom Comparer; if you control the implementation of type, you can also consider using an Exporter, AllowUnexported, or cmpopts.IgnoreUnexported
```

Reply:
Oops, my bad, of course DeepEqual can't use direct `==` comparison as it wouldn't be able to make a field-by-field diff without recursively comparing fields. Setting a custom comparer for `ExtDNSEntry` would also prevent it from doing a field-by-field diff of its fields. [The _documented_ right answer is `cmpopts.EquateComparable(netip.Addr{})`](https://pkg.go.dev/github.com/google/go-cmp@v0.6.0/cmp/cmpopts#EquateComparable)

Reply:
Ahh - that's handy, thank you ... it's new in go-cmp 0.6.0 - even more handily, that's in master as of last week. So, I'll rebase and do that.